### PR TITLE
cleanup of work queue API documentation and minor behavior changes

### PR DESCRIPTION
**Pull request by pabigot**
This is an alternative to #28579 that does not change the existing behavior related to cancelling delayed work items, but does update the documentation to describe that behavior as accurately as possible, including more about its limitations.

It also fixes a minor race condition and extends the tests to improve coverage and verify the return values in cases where the states can be observed.  This increases coverage of kernel/work_q to 100%.

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]

**Review by Vudentz — APPROVED** [review]
LGTM

File: kernel/work_q.c [hunk 1]
```diff
@@ -132,12 +128,12 @@ int k_delayed_work_submit_to_queue(struct k_work_q *work_q,
...
 {
-	if (!work->work_q) {
-		return -EINVAL;
-	}
-
 	k_spinlock_key_t key = k_spin_lock(&lock);
-	int ret = work_cancel(work);
+	int ret = -EINVAL;
+
+	if (work->work_q) {
```
Comment (nashif) [inline kernel/work_q.c:136]: You need to explicitly compare pointer with NULL (misra 14.4)

Comment (pabigot) [reply kernel/work_q.c]: done

**Comment by pabigot** [comment]
Added timer alignments to the delayed work tests to avoid  intermittent failures on mps2_an521.

File: kernel/work_q.c (hunk 1 above)
Comment (pabigot) [reply kernel/work_q.c]: @nashif please revisit, thanks.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**